File: testSpark/Python/leerDatos.py
%pyspark
from pyspark.sql import SparkSession
from pyspark.sql.functions import (col, concat, count, countDistinct,
                                   current_timestamp, date_format, desc, floor,
                                   lag, lit, monotonically_increasing_id, sum,
                                   udf, unix_timestamp)
from pyspark.sql.types import FloatType, StringType
from pyspark.sql.window import Window
from pyspark.sql import SQLContext
import time

#Clase APP, lee un csv a RDD; lo convierte a dataframe y le añade la columna Total
class APP:
    
    data = None
    def loadCSV():
        #Leemos CSV a RDD
        test = sqlContext.read.csv("/home/alex/Escritorio/CodigoYDatos/billing.csv",header=True)
        #De RDD a Dataframe
        df = test.toDF('InvoiceNo','StockCode','Description','Quantity','InvoiceDate', 'UnitPrice', 'CustomerID','Country')
        #Cambio del tipo de variable de algunas columnas(por defecto todas son string)
        df = df.withColumn('UnitPrice', df["UnitPrice"].cast("float"))
        df = df.withColumn('Quantity', df["Quantity"].cast("int"))
        df = df.withColumn('CustomerID', df["CustomerID"].cast("int"))
        #Añadimos columna Total(unitPrice*Quantity)
        dfmod = df.withColumn('Total', df["UnitPrice"] * df["Quantity"])
        return dfmod
        
    #Metodo que devuelve un dataframe a partir de un csv
    def get_datos():
        #Leemos CSV a RDD
        test = sqlContext.read.csv("/home/alex/Escritorio/CodigoYDatos/billing.csv",header=True)
        #De RDD a Dataframe
        df = test.toDF('InvoiceNo','StockCode','Description','Quantity','InvoiceDate', 'UnitPrice', 'CustomerID','Country')
        #Cambio del tipo de variable de algunas columnas(por defecto todas son string)
        df = df.withColumn('UnitPrice', df["UnitPrice"].cast("float"))
        df = df.withColumn('Quantity', df["Quantity"].cast("int"))
        #Añadimos columna Total(unitPrice*Quantity)
        dfmod = df.withColumn('Total', df["UnitPrice"] * df["Quantity"])
        return dfmod
    
    #Otra manera de leer csv a dataframe
    def get_datos2():
        test = sc.textFile("/home/alex/Escritorio/CodigoYDatos/billing.csv").map(lambda line: line.split(","))
        test_df = test.toDF(['InvoiceNo','StockCode','Description','Quantity','InvoiceDate', 'UnitPrice', 'CustomerID','Country'])
        return test_df

      

File: testSpark/Scala/leerDatos.scala
%spark
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions
import org.apache.spark.sql.types
import org.apache.spark.sql.expressions.Window
import org.apache.spark.sql.functions
import org.apache.spark.sql.DataFrame
import org.apache.spark.sql.functions.year

    
    def get_datos2(): DataFrame =
    {
        var data2 = sqlContext.read.csv("/home/alex/Escritorio/CodigoYDatos/billing.csv")
        var df_aux = data2.toDF("InvoiceNo","StockCode","Description","Quantity","InvoiceDate", "UnitPrice", "CustomerID","Country")
        return df_aux
    }
    def get_datos(): DataFrame = 
    {
        var data = sqlContext.read.format("com.databricks.spark.csv")
                            .option("header","true")
                            .option("inferSchema","true")
                            .load("/home/alex/Escritorio/CodigoYDatos/billing.csv")
        data = data.withColumn("UnitPrice",data("UnitPrice").cast("Float"))
        data = data.withColumn("Quantity",data("Quantity").cast("Int"))
        data = data.withColumn("CustomerID",data("CustomerID").cast("Int"))
        //data = data.withColumn("InvoiceDate",data("InvoiceDate").cast("Date"))
        var df = data.withColumn("Total", data("UnitPrice") * data("Quantity"))
        return df                 
    }
     /* Otra manera de leer Dataframes     */                 
    def get_datos3(): DataFrame = 
    {
       var data3 = spark.read.format("csv")
                                .option("header", "true") //reading the headers
                                .option("mode", "DROPMALFORMED")
                                .load("/home/alex/Escritorio/CodigoYDatos/billing.csv")
        return data3
    }
    
    def get_datos4(): DataFrame = 
    {
        var data = sqlContext.read.format("com.databricks.spark.csv")
                            .option("header","true")
                            .option("inferSchema","true")
                            .load("/home/alex/Escritorio/CodigoYDatos/billing.csv")
        return data                 
    }
var c = get_datos()
c.show()
